docs/rfcs/004-issueish-list.md

```diff
@@ -0,0 +1,98 @@
+* Title, truncated if necessary
+* Terse relative timestamp (1d, 2h, 30m)
+* Pull request state: open, closed, merged
+* Status check summary
+
+_TBD: Tile design_
+
+Clicking on a list item opens an issueish pane item for the chosen issueish in the same pane container as the parent component (by default, the right dock). If the issuish pane item is already open, it is activated instead.
+
+### Issueish Pane Item: Pull Request
+
+For a pull request, the issueish pane shows:
+
+* Author avatar
+* Title
+* Controls for actions:
+  * "Checkout" to fetch (if necessary) and check out the pull request. Only enabled if the current pull request is not the current one.
+  * "Browse" to launch a browser on the corresponding GitHub page.
+  * "Merge" to merge the pull request on GitHub if it is open.
+  * "Close" to close the pull request, unmerged, if it is open.
+  * "Re-open" to re-open a pull request if it is closed.
+* Pull request overview details: commit count, files changed.
+* Full description as rendered markdown.
+* Reaction emoji and counts.
+* Details and links for all status checks.
+
+_TBD: Issueish pane item design_
+
+## Drawbacks
+
+This does not offer a mechanism to create _new_ pull requests, which we have now. We also lose timeline events on the pull request.
+
+## Rationale and alternatives
+
+Our current GitHub panel focuses on showing you stuff about _the pull request that's associated with your current branch._ The problem is, it's difficult to unambiguously determine that in the general case.
+
+The first thing you see today when you open the GitHub panel on the `master` branch of an active repository is not very helpful:
+
+![wat](https://user-images.githubusercontent.com/17565/40857603-99b92304-65a9-11e8-986e-0f14290bda8a.png)
+
+This is a list of _all pull requests on GitHub that have a head ref called "master", from any head repository_. You can then "pin" any of them to see that pull request's details. This isn't useful on master and it's unclear to users what this is supposed to accomplish. Pinning was intended to be an infrequent edge case when we couldn't find the right PR for a given ref, not the first interaction you have with the package.
+
+Showing a PR list instead provides a uniform, more easily understood entry point to the package's GitHub functionality, and paves the way to other pull-request-focused activities in the future. "All open PRs" seems like a reasonable starting point, and "current PRs" preserves the ability to take advantage of your local editor context.
+
+With that said, the choices for the specific lists we show are a bit arbitrary. We'll need to research and iterate on them quite a bit to find what's most useful for the most people, but for now we need to start with something.
+
+## Unresolved questions
+
+### Before RFC merge:
+
+- [ ] How can we still offer "push/publish+new pull request"?
+- [ ] What else from the existing issueish pane should we keep? Comments, timeline events?
```

Comment:
The main reason I'd consider taking these out is because they don't really add anything to just... looking at the PR on .com.

Reply:
I updated the RFC to keep the existing center pane when clicking on "Conversation". I'm not sure how valuable it is, but since it's already built, we can keep it for now?

Other things from the current GitHub panel:

- Labels? Might still be ok to keep them?
- The detailed CI status?
  ![image](https://user-images.githubusercontent.com/378023/41099539-cf0c9e38-6a99-11e8-9c10-2139d2d490bb.png)
  It takes up quite some space and I'm not sure how valuable it is to see which one is failing. In the current mockup there is the "Checks", but I think that's a new feature that not every repo uses yet. Then the "Able to merge" would include CI, but also if a branch has conflicts or there if still a review is needed. If any of these fail, you would have to click on it and see on .com what it is and take further actions.

Reply:
> I updated the RFC to keep the existing center pane when clicking on "Conversation". I'm not sure how valuable it is, but since it's already built, we can keep it for now?

:+1: I like it for:

* Seeing the full conversation without Massive Endless Cramped Scrolling
* Detailed checks and build links
* Being able to drag and drop around as its own thing, so you can open more than one at once - like to keep an eye on one PR to see if it's green while you're working on another one

> It takes up quite some space and I'm not sure how valuable it is to see which one is failing.

One of my most-used features, personally 😁 It's a great way to keep an eye on builds while continuing to work, and to jump right to the problem when there is one.

Maybe we could come up with a more compact representation (eventually, if not here)?